### PR TITLE
feat(queries): Overhaul Queries Tab 

### PR DESCRIPTION
**Pull request by jjoyce0510**
## Summary

In this PR we improve the Dataset Queries Tab with new capabilities. Specifically, we enable creating, documenting, easily sharing custom queries for an Entity. 

There are now 2 sections on the Queries Tab: Highlighted Queries and Recent Queries. 

Highlighted queries are manually controlled - created and updated by humans. Recent Queries are those which were actually executed against the Dataset and automatically extracted (supported for BigQuery, Redshift, and Snowflake). 

Using this feature requires the `Edit Queries` privilege for a Dataset. 

## Changes

- [x] Entity + Aspect Models
- [x] GraphQL Schema 
- [x] GraphQL Resolvers
- [x] GraphQL Resolver Tests
- [x] GraphQL Type 
- [x] GraphQL Type Test
- [x] New Metadata Privileges
- [x] Default Policies Updated
- [x] GraphQL Query Service
- [x] GraphQL Query Service Tests
- [x] React GraphQL Queries
- [x] React Components 
- [x] React Caching 
- [x] React Usage Tracking 
- [x] Cypress Integration Tests
- [x] Sample / Bootstrap Data
- [x] Update Privileges Docs
- [ ] Smoke Tests (Next PR)
- [ ] Feature Documentation (Next PR)

## Status

In progress. Finalizing integration tests. 

## Screenshots

Screen Shot 2023-02-09 at 7 33 09 PM
Screen Shot 2023-02-09 at 7 33 44 PM (1)
Screen Shot 2023-02-10 at 10 26 09 AM (1)

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/authorization/AuthorizationUtils.java [hunk 1]
```diff
@@ -112,6 +113,35 @@ public static boolean canManageGlobalViews(@Nonnull QueryContext context) {
...
+
+  public static boolean canUpdateQuery(@Nonnull List<Urn> subjectUrns, @Nonnull QueryContext context) {
+    // Currently - you only need permission to edit an entity's queries to update any query.
+    return canEditEntityQueries(subjectUrns, context);
+  }
+
+  public static boolean canDeleteQuery(@Nonnull Urn entityUrn, @Nonnull List<Urn> subjectUrns, @Nonnull QueryContext context) {
+    // Currently - you only need permission to edit an entity's queries to remove any query.
+    return canEditEntityQueries(subjectUrns, context);
+  }
```
Comment (chriscollins3456) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/authorization/AuthorizationUtils.java:143]: makes sense to me!

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Queries/Query.tsx [hunk 2]
```diff
@@ -1,55 +1,72 @@
...
+                    showDetails={showDetails}
+                />
+            )}
+            {showEditQueryModal && (
+                <QueryBuilderModal
+                    initialState={{ urn: urn as string, title, description, query }}
+                    onSubmit={onEditSubmitted}
+                    onClose={() => setShowEditQueryModal(false)}
+                />
+            )}
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Queries/Query.tsx:69]: this whole queries frontend is beautifully packaged and organized into nice tight components. thank you!

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Queries/utils/filterQueries.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Query } from '../types';
+
+/**
+ * Filter queries by a search string. Compares name, description, and query statement.
+ *
+ * @param filterText the search text
+ * @param queries the queries to filter
+ */
+export const filterQueries = (filterText, queries: Query[]) => {
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Queries/utils/filterQueries.ts:9]: nit: could add a unit test for this pure fxn. but if we want to get this out feel free to ignore!

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Queries/utils/getTopNQueries.ts [hunk 4]
```diff
@@ -1,24 +1,32 @@
 import { UsageAggregation, Maybe } from '../../../../../../../types.generated';
 
-export default function getTopNQueries(responseSize: number, buckets?: Maybe<UsageAggregation>[] | null) {
-    let response: (string | null)[] = [];
+type RecentQuery = {
+    query: string;
+    dateMs: number;
+};
+
+export default function getTopNQueries(n: number, buckets?: Maybe<UsageAggregation>[] | null): RecentQuery[] {
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Queries/utils/getTopNQueries.ts:8]: same comment here about test if we have time

**Review by chriscollins3456 — APPROVED** [review]
looking wonderful!

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Queries/Query.tsx (hunk 2 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/shared/tabs/Dataset/Queries/Query.tsx]: <3

**Pull request merged**